src/Messages/Attachments/TSAttachmentStream.m

```diff
@@ -34,10 +39,27 @@ - (instancetype)initWithPointer:(TSAttachmentPointer *)pointer
 
     _contentType = pointer.contentType;
     _isDownloaded = YES;
+    // TSAttachmentStream doesn't have any "incoming vs. outgoing"
+    // state, but this constructor is used only for new incoming
+    // attachments which don't need to be uploaded.
+    _isUploaded = YES;
 
     return self;
 }
 
+- (void)upgradeFromAttachmentSchemaVersion:(NSUInteger)attachmentSchemaVersion
```

Comment:
Pulling out the schema version upgrade logic to a method so that subclasses can override and add upgrade logic of their own.